### DISCUSSION TITLE
Correct KWh consumption in charge list

### DISCUSSION THREAD

**Discussion by lvenditti** [flobz/psa_car_controller#245] [Ideas]
Hi,
very compliment for your job. 
I am using it on a raspberry PI4 and it is working well. Have just a suggestion about consumption reported in charge list.
With the default value of 46 for battery power into file cars.json the consumption is under estimated.
I had to put 53 for this parameter in order to obtain real value (value taken from energy meter of my wallbox).

As example, if I charged from 53% to 79% I have 12.22 KWh as consumption on the charging list but the real consumption, taken from energy meter of wallbox, is 13.92 KWh.

I think this is due to charging loss of OBC.
So my suggestion is to use this value as default (or adapt script in the way it will consider the charging loss).

Thanks again for your great job!!

Cheers
Luigi

**Comment by flobz** [comment]
Yes that's a good idea. I will had a loss parameter to the calculation.

**Comment by DavidFFerreira** [comment]
@lvenditti the loss is variable, i did that, even I have a power metter control that have even a diferent value....soo I have 3 diferent values. dunno what is real or not lol

**Reply by lvenditti** [reply]
Yes, you right the loss is variable, it depend mainly on external temperature, on power used to charge and the current SOC%.

But I think that we can use for losses a value that could be a balance that will give a better value for consumption.
(as example I put 53 into cars.json file, instead of 46 and in this way I have better value for consumption)

This value could be inserted on a configuration file as different parameter so anyone will be able to adapt it to its need and leaving 46 as it is.

In my experience the value taken from wall box meter is very close to the value of energy meter in my house (from public energy network). Instead the value shown on charge list is about 11-12% less the real (payed) consumption.

Cheers
Luigi

**Reply by JHCD** [reply]
> In my experience the value taken from wall box meter is very close to the value of energy meter in my house (from public energy network). Instead the value shown on charge list is about 11-12% less the real (payed) consumption.

That's totally normal. The vehicle reports the net charge that has arrived in the battery, the wallbox reports the gross energy.
A load loss of only 11-12% is pretty good, for PSA vehicles you will find more information around 22% in the statistics

As an approximation, an indication of an average charge loss (e.g. 22%) would probably be sufficient. Alternatively, the (externally determined) gross load could be specified for each charging process.

e.g. source: http://www.praxis-elektroauto.de/e-mobilitaet/ladeverluste.html (German)

**Reply by lvenditti** [reply]
"A load loss of only 11-12% is pretty good, for PSA vehicles you will find more information around 22% in the statistics"

The 11-12% loss is calculated for my eC4. I am changing usually at 10-12 A with a wall box starting from 40-50% of SOC to 80% of SOC.

Cheers
Luigi

**Comment by DavidFFerreira** [comment]
srange. I have a smartplug that gave me the kw per minut, and what I see is a wast of 20%/30%, each charging have is own waste, the wast is not the same everytime
I have wast between chargers of 18% to 34%

**Reply by JHCD** [reply]
In addition to the question of how exactly such a smart plug is (my smart plugs alone have a deviation of ~ 5% compared to my calibrated meters), the charging loss depends on many factors: SoC start, SoC end, temperature, charging power (the lower the higher the loss) etc.

**Comment by lvenditti** [comment]
Actually is very hard to find a real calculation of loss (too many factors), but what we pay is the gross energy not the net energy consumed by the car.
In my experience, just modifying the "battery power" parameter into cars.json file from 46 to 55.8 I have a better estimation of gross energy need for the car per 100 Km.  It is not perfect, but with this trick, for in house charging, I have a good approximation of gross energy consumption.

Could be interesting which value you have to insert into 'battery power' parameter in order to have an approximation of gross energy (you could take this value from WB meter or public network meter) at least for one condition: charging by WB at 12 A with the car located in the box (T about 16 Celsius) and starting at 40% and ending at 80% of SOC.
In this way we could do some comparison for different PSA cars (E-C4 in my case).

Cheers.
